### DISCUSSION TITLE
How the list items are filtered by permission on the Content Item list page on the Admin UI?

### DISCUSSION THREAD

**Discussion by barthamark** [OrchardCMS/OrchardCore#10283] [Q&A] [answered]
I have a user who has EditContent permission for _Content Type A_ but only ViewContent permission for _Content Type B_. When I log in and go to _Admin UI > Content > Content Items_ list I only see content items of _Content Type A_. No filters are applied. I'd like to understand what happens under the hood. Where is the filtering by authorized items is done?

I tried to debug the OC code but I'm lost:

This is where the query is generated in AdminController.List: https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs#L239

This is where the content type filter is added to the query: https://github.com/OrchardCMS/OrchardCore/blob/5be72c8df3599df798cba510411115fe315494cf/src/OrchardCore.Modules/OrchardCore.Contents/Services/DefaultContentsAdminListQueryService.cs#L38

Now it becomes a YesSQL territory that I can't really debug effectively and don't really understand.

I see that the `FilterResult` object is coming from action parameter with custom model binding, but when I debug the model binder all I see is that it parses an empty string: https://github.com/OrchardCMS/OrchardCore/blob/339d74b251b53393dd510f8ad3027aa17d78500d/src/OrchardCore/OrchardCore.Contents.Core/Services/ContentItemFilterEngineModelBinder.cs#L30

I also read the inline documentation, this one might be a clue: https://github.com/OrchardCMS/OrchardCore/blob/5be72c8df3599df798cba510411115fe315494cf/src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs#L116 but I don't understand it.

I'm most probably missing something trivial here.

**Comment by Skrypt** [comment]
If a user has only ViewContent permission then he doesn't need to manage these content items. So, by default, in the admin UI he doesn't need to list these content items. Though, there is one permission under "OrchardCore.Contents" that allows to list any content items owned by other users : "List content item(s) owned by all users". This will allow to display any content items even if they don't have the right to edit them.

**Reply by barthamark** [reply]
I'm okay with this, but my question was about how it's implemented, where can I find the logic where the IQuery becomes populated with the types the user can edit. See my linked code lines as well, I found those, but none of them answers my question, unfortunately.

By the way, it's not correct:
> "List content item(s) owned by all users". This will allow to display any content items even if they don't have the right to edit them.

I have a role with this permission allowed and EditContent for one type and the user can only see items of this type.

**Reply by barthamark** [reply]
I found it in the meantime (see my answer). Based on the logic implemented there, it looks like it's intentional that even if the user has that permission they still can't see all the items. Anyway, thank you for your reply!

**Comment by barthamark** [comment, marked as answer]
Found it.

https://github.com/OrchardCMS/OrchardCore/blob/4f0a3c2d86ddee457f6d0d8fd9166ff4827a6d08/src/OrchardCore.Modules/OrchardCore.Contents/Services/DefaultContentsAdminListFilterProvider.cs#L127

**Reply by Skrypt** [reply]
I will think about it but maybe it should display them if the View permission is set. Maybe the confusion is that by default it is set on every content types even for the "Anonymous" role else they can't see these content items on the frontend. Of course, you can modify the DefaultContentsAdminListFilterProvider for your own needs too.

**Reply by Skrypt** [reply]
Maybe we need a permission "View in admin" which would be differenciated from "View" that implies "frontend".

**Reply by barthamark** [reply]
I think it'd be fine to list items even if they don't have permission to edit them only to view. It automatically hides the Edit button so there are no security issues with that. It might be worth creating an issue.

**Reply by Skrypt** [reply]
At this point I think it's just a matter of taste. I personally don't see a need to list content items I can't manage in the admin which is context is to manage content items ...

So, what is the added value in that case? Allowing to add comments on non-owned content items when there is an Audit Trail commenting module?

I'm not against the idea but again I need to say that currently if we add the permission with the ViewContent permission it won't work since it is used to display content items on the frontend (Anonymous role). So basically, everyone has the ViewContent permission because the Anonymous user has it.

Solution is to either refactor the way we use the ViewContent permission to be only for the admin or we add a ViewAdminContent permission which seems kind of awkward. I may be wrong, since it's been a while I worked on that part of the code but I'm pretty sure it will be a breaking change.
